src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala

```diff
@@ -681,13 +693,86 @@ class AstCreator(filename: String, global: Global) {
     Ast(NewLiteral().order(order).argumentIndex(order).code(x.toString).typeFullName("double"))
   }
 
+  def astForClassExpr(expr: ClassExpr, order: Int): Ast = {
+    val callNode = NewCall()
+      .name(Operators.fieldAccess)
+      .methodFullName(Operators.fieldAccess)
+      .dispatchType(DispatchTypes.STATIC_DISPATCH)
+      .code(expr.toString)
+      .argumentIndex(order)
+      .order(order)
+
+    val identifer = NewIdentifier()
+      .typeFullName("ANY")
+      .code(expr.getTypeAsString)
+      .lineNumber(line(expr))
+      .columnNumber(column(expr))
+      .argumentIndex(1)
+      .order(1)
+
+    val fieldIdentifier = NewFieldIdentifier()
+      .canonicalName("class")
+      .code("class")
+      .lineNumber(line(expr))
+      .columnNumber(column(expr))
+      .argumentIndex(2)
+      .order(2)
+
+    callAst(callNode, Seq(Ast(identifer), Ast(fieldIdentifier)))
+  }
+
+  def astForConditionalExpr(expr: ConditionalExpr, order: Int): Ast = {
+    val callNode = NewCall()
+      .name(Operators.conditional)
+      .methodFullName(Operators.conditional)
+      .dispatchType(DispatchTypes.STATIC_DISPATCH)
+      .code(expr.toString)
+      .argumentIndex(order)
+      .order(order)
```

Comment:
you should add the line number here

Reply:
Also a good point. I've updated this case now, but will do a more comprehensive check to see if line/col numbers exist and are what we expect them to for everything supported so far.